lib/dataset.js

```diff
@@ -244,78 +247,94 @@ Dataset.prototype.stream = function (options) {
 // This is necessary because all pipeline functions are synchronous
 // and to avoid back pressure during streaming out.
 //
-Dataset.prototype.save = thenify(function (path, options, done) {
-  options = options || {};
+Dataset.prototype.save = thenify(function (path, options = {}, done) {
   if (arguments.length < 3) done = options;
   path = path.replace(/\/+$/, '');  // Trim trailing slashes (confusing for S3)
-  var opt = {
+  const opt = {
     gzip: options.gzip,
     parquet: options.parquet,
     stream: options.stream,
     csv: options.csv,
     path: path,
     _preIterate: function (opt, wc, p) {
-      var suffix = opt.gzip ? '.gz' : opt.parquet ? '.parquet' : '';
```

Comment:
Do not change the suffix assignment logic in this PR. It should be proposed and discussed separately. I'm also missing proper regression tests for file extension handling.